Adds the installed message phrase for voice analyzers to logging (bonus runtime fix)

san7890 opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

Hey there,

Sometimes, it's CBT to figure out what exactly made a bomb go off, especially when a voice analyzer is involved. Now, when a voice analyzer is involved in TTV Bomb/Grenade explosions (already logged), it will also output the recorded phrase when present.

To do this, I also had to fix this following runtime, which would occur since voice analyzers didn't pass `bomber`:

![image](https://user-images.githubusercontent.com/34697715/191871302-58f6cdad-e1c8-4f45-be40-24ab03143098.png)

Also, that would result in the whole proc failing and the voice analyzer _never_ exploding the TTV. That's fucked up! I had to fix it in this PR because otherwise, the voice analyzer would _never_ explode the bomb, and what's the point of all that logging?

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game

For chemical grenades:
![image](https://user-images.githubusercontent.com/34697715/191642190-94646a7c-1891-4815-9676-1f2ffad71faf.png)

For TTV bombs (forgot to get a screenshot, so have an excerpt of the game.log from that round)
```txt
[2022-09-22 00:35:08.128] GAME: Bomb valve opened in Ordnance Testing Lab (191,102,2) with the voice analyzer attached by <a href='?priv_msg=san7890'>San7890</a>/(Katie Yossarian) with the following message: "Deez nuts." - Last touched by: <a href='?priv_msg=san7890'>San7890</a>/(Katie Yossarian)
[2022-09-22 00:35:09.842] GAME: /obj/item/tank/internals/oxygen exploded with a power of 231.346 and a mix of TEMP=1544.11, MOL=197.675, VOL=140 o2=179.56;plasma=16.4251;co2=1.26777;water_vapor=0.422589;
[2022-09-22 00:35:09.842] GAME: Explosion with size (5, 10, 20, 0) in (Ordnance Testing Lab (191,102,2)).  Possible cause: the oxygen tank. Last fingerprints: *null*.
```

Keep in mind that I tweaked the messages a bit since I tested to ensure they worked in game.

Just allows for better administration rather than having to scroll up and down to figure out what specific phrase could have set it off.

<!-- Argue for the merits of your changes and how they benefit the game, especially if they are controversial and/or far reaching. If you can't actually explain WHY what you are doing will improve the game, then it probably isn't good for the game in the first place. -->

## Changelog

<!-- If your PR modifies aspects of the game that can be concretely observed by players or admins you should add a changelog. If your change does NOT meet this description, remove this section. Be sure to properly mark your PRs to prevent unnecessary GBP loss. You can read up on GBP and it's effects on PRs in the tgstation guides for contributors. Please note that maintainers freely reserve the right to remove and add tags should they deem it appropriate. You can attempt to finagle the system all you want, but it's best to shoot for clear communication right off the bat. -->

:cl:
admin: When a voice analyzer is attached to a bomb or a chemical grenade, the respective logs to game as well as the message_admins will contain the recorded phrase on that voice analyzer.
fix: Voice analyzers should now properly trigger TTVs to blow up.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

san7890 committed:
Adds the installed message phrase for voice analyzers to logging

Hey there,

Sometimes, it's CBT to figure out what exactly made a bomb go off, especially when a voice analyzer is involved. Now, when a voice analyzer is involved in TTV Bomb/Grenade explosions (already logged), it will also output the recorded phrase when present.

To do this, I just span up a quick `isvoice()` macro define to ensure that we would get the variable, and it would just append the message that we were already spitting out into logs. I rewrote how grenade logs currently operate a bit, let me know if I accidentally omitted something critical.

Mothblocks commented on code/__DEFINES/is_helpers.dm:
```diff
@@ -235,6 +235,8 @@ GLOBAL_LIST_INIT(turfs_openspace, typecacheof(list(
 
 #define isprox(O) (istype(O, /obj/item/assembly/prox_sensor))
 
+#define isvoice(O) (istype(O, /obj/item/assembly/voice))
```
this name is bad. please just use istype directly we need to stop making more of these when theyre only used by like 2 things

san7890 committed:
the commit that uses istype directly

san7890 committed:
I also fix a newly uncovered runtime here

I'll update the PR regarding this but apparently if you triggered the bomb using a voice analyzer, this would runtime since there was no "bomber". This resulted in getting a runtime where you could not execute a "log" on null, and the bomb wouldn't explode I think.
